frontend/class-handle-404.php

```diff
@@ -64,6 +64,8 @@ protected function is_feed_404( $handled ) {
 		$wp_query->is_feed = false;
 		$this->set_404();
 
+		add_filter( 'old_slug_redirect_url', '__return_false' );
+
```

Comment:
It's possible that `redirect_guess_404_permalink` find post if names are similar. Could you try to add:

```suggestion
add_filter( 'redirect_canonical', '__return_false' );

```